benchmarks/union_find/union_find.cpp

```diff
@@ -0,0 +1,178 @@
+                         "ArborX::Benchmark::edges"),
+      num_edges);
+
+  Kokkos::Random_XorShift1024_Pool<ExecutionSpace> rand_pool(1984);
+  Kokkos::parallel_for(
+      "ArborX::Bechmark::init_edges",
+      Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, num_edges),
+      KOKKOS_LAMBDA(unsigned i) {
+        auto rand_gen = rand_pool.get_state();
+        do
+        {
+          edges(i) = {rand_gen.urand() % num_edges,
+                      rand_gen.urand() % num_edges};
+        } while (edges(i).source == edges(i).target); // no self loops
+        rand_pool.free_state(rand_gen);
+      });
+  return edges;
+}
+
+template <typename ExecutionSpace>
+Kokkos::View<UnweightedEdge *, typename ExecutionSpace::memory_space>
+buildEdges(DisallowLoops, ExecutionSpace const &exec_space, int num_edges)
+{
+  using MemorySpace = typename ExecutionSpace::memory_space;
+
+  Kokkos::Random_XorShift1024_Pool<ExecutionSpace> rand_pool(1984);
+  // Construct random permutation by sorting a vector of random values
+  Kokkos::View<int *, MemorySpace> random_values(
+      Kokkos::view_alloc(exec_space, Kokkos::WithoutInitializing,
+                         "ArborX::Benchmark::random_values"),
+      num_edges);
+  Kokkos::parallel_for(
+      "ArborX::Bechmark::init_random_values",
+      Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, num_edges),
+      KOKKOS_LAMBDA(int i) {
+        auto rand_gen = rand_pool.get_state();
+        random_values(i) = rand_gen.rand();
+        rand_pool.free_state(rand_gen);
+      });
+  auto permute = ArborX::Details::sortObjects(exec_space, random_values);
+
+  // Init edges in a random order
+  Kokkos::View<UnweightedEdge *, MemorySpace> edges(
+      Kokkos::view_alloc(exec_space, Kokkos::WithoutInitializing,
+                         "ArborX::Benchmark::edges"),
+      num_edges);
+  Kokkos::parallel_for(
+      "ArborX::Bechmark::init_edges",
+      Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, num_edges),
+      KOKKOS_LAMBDA(unsigned i) {
+        auto rand_gen = rand_pool.get_state();
+        edges(permute(i)) = {rand_gen.urand() % (i + 1), i + 1};
+        rand_pool.free_state(rand_gen);
+      });
+  return edges;
+}
+
+template <typename ExecutionSpace>
+auto buildUnionFind(ExecutionSpace const &exec_space, int n)
+{
+  using MemorySpace = typename ExecutionSpace::memory_space;
+
+  Kokkos::View<int *, MemorySpace> labels(
+      Kokkos::view_alloc(Kokkos::WithoutInitializing,
+                         "ArborX::Benchmark::labels"),
+      n);
+  ArborX::iota(exec_space, labels);
+#ifdef KOKKOS_ENABLE_SERIAL
+  if constexpr (std::is_same_v<ExecutionSpace, Kokkos::Serial>)
+    return ArborX::Details::UnionFind<MemorySpace, /*DoSerial*/ true>(labels);
+  else
+#endif
+    return ArborX::Details::UnionFind<MemorySpace, /*DoSerial*/ false>(labels);
+}
+
+template <typename ExecutionSpace, typename Tag>
+void BM_union_find(benchmark::State &state)
```

Comment:
Please drop that `ExecutionSpace` template parameter.
It yields names like `BM_union_find<Kokkos::DefaultExecutionSpace,AllowLoops>` where the execution space info is pointless